pwiz_tools/Skyline/Model/ModificationMatcher.cs

```diff
@@ -75,7 +77,7 @@ public override bool MoveNextSequence()
                 return false;
             // Skip sequences that can be created from the current settings.
             // Check first if the sequence has any modifications, because creating doc nodes is expensive
-            while (!HasMods(_sequences.Current) ||
+            while (!HasModsChecked(_sequences.Current) ||

```

Comment:
This function could have been simpler if the condition for the "while" were "MoveNextSingleSequence".

I would probably rewrite this function as:
```
        public override bool MoveNextSequence()
        {
            while (MoveNextSingleSequence())
            {
                bool hasMods = HasMods(_sequences.Current);
                HasSeenMods |= hasMods;
                // Skip sequences that can be created from the current settings.
                // Check first if the sequence has any modifications, because creating doc nodes is expensive
                if (hasMods)
                {
                    if (CreateDocNodeFromSettings(new Target(_sequences.Current), null, DIFF_GROUPS, out _) == null)
                    {
                        return true;
                    }
                }
            }

            return false;
        }
```